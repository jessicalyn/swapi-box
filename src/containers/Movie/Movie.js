import React, { Component } from 'react'
import './Movie.css'
import { connect } from 'react-redux'

export class Movie extends Component {
  render() {
    const { date, title, crawl } = this.props.movie
    return(
      <div className="movie-container">
        <section className="star-wars">
          <div className="crawl">
          <div className="title">
            <h1>{ title }</h1>
          </div>
          <p>{ crawl }</p>
          <p>{ date }</p>
          </div>
        </section>
      </div>
    )
  }
}
export const mapStateToProps = (state) => ({
  movie: state.movie
})

export default connect(mapStateToProps)(Movie)