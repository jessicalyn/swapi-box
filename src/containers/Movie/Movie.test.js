import React from 'react'
import { Movie, mapStateToProps } from './Movie'
import { shallow } from 'enzyme'

describe('Movie', () => {
  describe('Movie', () => {
    let wrapper

    beforeEach(() => {
      const mockMovie = [{
        title: "Return of the Jedi", 
        year: 2019, 
        crawl: "This is the movie text"
      }]
      wrapper = shallow(<Movie movie={ mockMovie } />)
    });

    it('should match the snapshot with all data passed in', () => {
      expect(wrapper).toMatchSnapshot()
    });

  });

  describe('mapStateToProps', () => {
    it('should return an object with a movie', () => {
      const mockState = {
        error: "",
        isLoading: "",
        movie: [{
          title: "Return of the Jedi", 
          year: 2019, 
          crawl: "This is the movie text"
        }]
      }
      const expected = { movie: [{
        title: "Return of the Jedi", 
        year: 2019, 
        crawl: "This is the movie text"
      }]}

      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  })
})