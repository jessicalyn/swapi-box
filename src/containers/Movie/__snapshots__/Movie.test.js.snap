// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Movie should match the snapshot with all data passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Movie
    movie={
      Array [
        Object {
          "crawl": "This is the movie text",
          "title": "Return of the Jedi",
          "year": 2019,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="star-wars"
      >
        <div
          className="crawl"
        >
          <div
            className="title"
          >
            <h1 />
          </div>
          <p />
          <p />
        </div>
      </section>,
      "className": "movie-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="crawl"
        >
          <div
            className="title"
          >
            <h1 />
          </div>
          <p />
          <p />
        </div>,
        "className": "star-wars",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="title"
            >
              <h1 />
            </div>,
            <p />,
            <p />,
          ],
          "className": "crawl",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1 />,
              "className": "title",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h1",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "section",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="star-wars"
        >
          <div
            className="crawl"
          >
            <div
              className="title"
            >
              <h1 />
            </div>
            <p />
            <p />
          </div>
        </section>,
        "className": "movie-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="crawl"
          >
            <div
              className="title"
            >
              <h1 />
            </div>
            <p />
            <p />
          </div>,
          "className": "star-wars",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title"
              >
                <h1 />
              </div>,
              <p />,
              <p />,
            ],
            "className": "crawl",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1 />,
                "className": "title",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "h1",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "section",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
